### DISCUSSION TITLE
execute lldb-vscode-11 commands?

### DISCUSSION THREAD

**Discussion by sharksforarms** [mfussenegger/nvim-dap#381] [Q&A] [answered]
Hey all,

I'm seeking to run lldb commands such as `parray 10 ptr` and more, but where is this suppose to be run? When I open the repl, I get a `dap > ` repl but lldb commands don't seem to be able to run in here. I have tried setting `runInTerminal = true` but that doesn't seem to do anything.

Here is my configuration:

```
dap.adapters.lldb = {
  type = 'executable',
  command = '/usr/bin/lldb-vscode-11',
  name = "lldb",
}

dap.configurations.cpp = {
  {
    name = "Launch",
    type = "lldb",
    request = "launch",
    program = function()
      return vim.fn.input('Path to executable: ', vim.fn.getcwd() .. '/', 'file')
    end,
    cwd = '${workspaceFolder}',
    stopOnEntry = false,
    args = {},

    -- if you change `runInTerminal` to true, you might need to change the yama/ptrace_scope setting:
    --    echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope
    -- Otherwise you might get the following error:
    --    Error on launch: Failed to attach to the target process
    -- But you should be aware of the implications:
    -- https://www.kernel.org/doc/html/latest/admin-guide/LSM/Yama.html
    runInTerminal = true,
  },
}
```

**Comment by theHamsta** [comment]
`lldb-vscode` is not `lldb`. As far as I know (might have changed since the last time I checked), there is no way to send lldb commands to the linked lldb instance. For this feature, you probably have to file a PR at to the LLVM-project. The best resource for lldb-vscode is mostly grepping it's source code: https://github.com/llvm/llvm-project/tree/main/lldb/tools/lldb-vscode The repl in dap can only do the same things that you could do in VSCode debug terminal.

I would love to also run the CLI commands. I even didn't manage to run Python commands that would allow pretty-printing for Rust types.

**Comment by vogelsgesang** [comment, marked as answer]
You can send lldb commands [by prefixing them with "`"](https://github.com/llvm/llvm-project/blob/842fa38dbeb70dcae992f8db7e5ba2d7f954f29a/lldb/tools/lldb-vscode/lldb-vscode.cpp#L1216).

For example, if you enter
```
`bt
```
on the `dap>` repl, you will get a backtrace printed by lldb

Afaict, this has been supported since [the first commit](https://github.com/llvm/llvm-project/commit/2f5cf8511a3f0ad97db2acf51c6c9ea0ddc9da92) of `lldb-vscode`

**Reply by theHamsta** [reply]
@vogelsgesang wow, nice! I didn't know that. Can you add this to the wiki?

**Reply by vogelsgesang** [reply]
Good point, I added a sentence about this to the "[Debug Adapter Installation](https://github.com/mfussenegger/nvim-dap/wiki/Debug-Adapter-installation#ccrust-via-lldb-vscode)" page on the wiki.
Not sure if there is a better place to put this. It seemed to be easiest to find in that location.

**Reply by AlexXandreE** [reply]
Hi, sorry to "reopen" this after so long, but shouldn't be possible to do with custom commands?
I know the "append" is incorrect and only prints to repl, but is there a way to make it work?

```
repl.commands = vim.tbl_extend('force', repl.commands, {
    custom_commands = {
        ['.trace'] = function(variable)
            dap.repl.append('`bt ' .. variable)
        end
    }
})
```

**Reply by mfussenegger** [reply]
Text written to the repl via `dap.repl.append` doesn't get re-evaluated. You'd have to call `session:evaluate('`bt ' .. variable, evaluate_handler)` and re-implement your own evaluate handler. See https://github.com/mfussenegger/nvim-dap/blob/f4a3be57f61893cffa1e22aa5e1e7bded495fcf2/lua/dap/repl.lua#L128-L141 for the default implementation
